docs/DeveloperGuide.md

```diff
@@ -739,6 +739,117 @@ testers are expected to do more *exploratory* testing.
+    2. Test case: `filter t/friends` <br>
+       Expected: Contact list displays only contacts with a tag called `friends`.
+
+2. Filter by multiple tags
+
+    1. Prerequisites: Contact list should contain contacts with both tags called `friends` and `classmastes`.
+
+    2. Test Case: `filter t/friends t/classmates` <br>
+       Expected: Contact list displays only contacts with both tags called `friends` and `classmastes`.
+
+### Search by schedule
+
+1. Search for schedules after a starting time
+
+    1. Prerequisites: Some contacts should have schedules with datetime after `2024-09-09 18:00`.
+    2. Test case: `search b/2024-09-09 18:00` <br>
+       Expected: Contact list displays the contacts with schedules after `2024-09-09 18:00`.
+
+2. Search for schedules before an end time
+
+    1. Prerequisites: Some contacts should have schedules with datetime before `2024-11-11 18:00`.
+
+    2. Test Case: `search en/2024-11-11 18:00` <br>
+       Expected: Contact list displays the contacts with schedules before `2024-11-11 18:00`.
+
+3. Search for schedules between a starting and an end time
+
+    1. Prerequisites: Some contacts should have schedules between `2024-09-09 18:00` and `2024-11-11 18:00`.
+
+    2. Test Case: `search b/2024-09-09 18:00 en/2024-11-11 18:00` <br>
+       Expected: Contact list displays the contacts with schedules between `2024-09-09 18:00` and `2024-11-11 18:00`.
+
+### renameTags
+
+1. rename existing tags to a new tag
+
+    1. Prerequisites: Some contacts should have tags called `friends`.
+    2. Test case: `renameTag ot/friends nt/bestFriends` <br>
+       Expected: Contact list updates the contact's tags to `bestFriends`.
+
+2. rename tags that don't exist
+
+    1. Prerequisites: None of the contacts has a tag called `classmates`.
+    2. Test case: `renameTag ot/classmates nt/bestFriends` <br>
+       Expected: The command fails with an error thrown.
+
+### Adding social media
+
+1. Adding a social media account to a contact
+
+    1. Prerequisites: The list contains at least one contact.
+    2. Test case: `socialMedia 1 fb/John` <br>
+       Expected: The contact will be updated showing a facebook account named John.
+
+2. Adding multiple social media account to a contact
+
+    1. Prerequisites: The list contains at least one contact.
+    2. Test case: `socialMedia 1 fb/John ig/John` <br>
+       Expected: The contact will only be updated showing an instagram account named John but not facebook account.
+
+### Restore
+
+1. Restore the last person deleted
+
+    1. Prerequisites: The person has to be deleted within the same section.
+    2. Test case: `restore` <br>
+       Expected: The deleted person is restored.
+
+2. Deleted person is added back before using `restore`
+
+    1. Prerequisites: A person is deleted in the same section and added back with the `add` command
+    2. Test case: `restore` <br>
+       Expected: No change.
```

Comment:
I think this portion can be clearer.
I tried this sequence:
1. add n/testrestore p/12345678
2. delete (added person index)
3. add n/testrestore p/12345678
4. restore
the output is that the person exists already

Similarly, if we delete a person, then edit another contact to be the deleted person's name, the restore command will show that the person exists in the contact book